control-plane/csi-controller/src/main.rs

```diff
@@ -97,10 +97,27 @@ pub async fn main() -> Result<(), String> {
         CsiControllerConfig::get_config().rest_endpoint()
     );
 
-    server::CsiServer::run(
-        args.value_of("socket")
-            .expect("CSI socket must be specfied")
-            .to_string(),
-    )
-    .await
+    let mut signal_term = tokio::signal::unix::signal(tokio::signal::unix::SignalKind::terminate());
+    let mut signal_int = tokio::signal::unix::signal(tokio::signal::unix::SignalKind::interrupt());
+
+    match (signal_term.as_mut(), signal_int.as_mut()) {
+        (Ok(termsig), Ok(intsig)) => {
+            tokio::select! {
+                _srv =  server::CsiServer::run(args.value_of("socket")
+                    .expect("CSI socket must be specfied")
+                    .to_string()) => {
+                    Ok(())
+                }
+                _evt = termsig.recv() => {
+                        opentelemetry::global::force_flush_tracer_provider();
+                        Ok(())
+                    }
+                _evt = intsig.recv() => {
+                        opentelemetry::global::force_flush_tracer_provider();
+                        Ok(())
+                    }
+            }
+        }
+        _ => Err("failed to add signal handlers".to_string()),
```

Comment:
If the signal handlers are a requisite, then you can do eg .expect("Failed to register SIGTERM handler") on line 100 and then we'll know exactly which one failed.

Reply:
Done https://github.com/mayadata-io/mayastor-control-plane/pull/158/commits/08896f39ff98e16a05b6014492cc5a70398050d4